modules/openapi-generator/src/main/java/org/openapitools/codegen/OpenAPINormalizer.java

```diff
@@ -0,0 +1,427 @@
+     *
+     * @param openAPI OpenAPI
+     * @param rules a map of rules
+     */
+    public OpenAPINormalizer(OpenAPI openAPI, Map<String, String> rules) {
+        this.openAPI = openAPI;
+
+        if (rules == null) {
+            return;
+        } else {
+            this.rules = rules;
+        }
+    }
+
+    /**
+     * Gets the rule.
+     *
+     * @return a map of rules
+     */
+    public Map<String, String> getRules() {
+        return this.rules;
+    }
+
+    /**
+     * Sets the rule.
+     *
+     * @param rules a map of rules
+     */
+    public void setRules(Map<String, String> rules) {
+        this.rules = rules;
+    }
+
+    /**
+     * Returns true if the rule is enabled.
+     *
+     * @param rule name of the rule
+     * @return boolean
+     */
+    public boolean isRuleEnabled(String rule) {
+        if ("true".equalsIgnoreCase(this.rules.get(rule))) {
+            // true or false
+            return true;
+        } else if (StringUtils.isNotEmpty(this.rules.get(rule))) {
+            // string value, e.g. pascal, snake, original
+            return true;
+        } else if (StringUtils.isEmpty(this.rules.get(rule))) {
+            return false;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Sets the rule.
+     *
+     * @param key   rule name
+     * @param value rule value
+     */
+    public void setRule(String key, String value) {
+        this.rules.put(key, value);
+    }
+
+    /**
+     * Normalizes the OpenAPI input, which may not perfectly conform to
+     * the specification.
+     */
+    void normalize() {
+        if (rules == null || rules.isEmpty()) {
+            return;
+        }
+
+        if (this.openAPI.getComponents() == null) {
+            this.openAPI.setComponents(new Components());
+        }
+
+        if (this.openAPI.getComponents().getSchemas() == null) {
+            this.openAPI.getComponents().setSchemas(new HashMap<String, Schema>());
+        }
+
+        normalizePaths();
+        normalizeComponents();
+    }
+
+    /**
+     * Normalizes inline models in Paths
+     */
+    private void normalizePaths() {
+        Paths paths = openAPI.getPaths();
+        if (paths == null) {
+            return;
+        }
+
+        for (Map.Entry<String, PathItem> pathsEntry : paths.entrySet()) {
+            PathItem path = pathsEntry.getValue();
+            List<Operation> operations = new ArrayList<>(path.readOperations());
+
+            // Include callback operation as well
+            for (Operation operation : path.readOperations()) {
+                Map<String, Callback> callbacks = operation.getCallbacks();
+                if (callbacks != null) {
+                    operations.addAll(callbacks.values().stream()
+                            .flatMap(callback -> callback.values().stream())
+                            .flatMap(pathItem -> pathItem.readOperations().stream())
+                            .collect(Collectors.toList()));
```

Comment:
🤔 We don't really need to create this temporary list, instead we can use `Stream.concat(path.readOperations().stream, callbacks.values().stream().flatMap...)`